pkg/collector/spec.go

```diff
@@ -271,16 +271,40 @@ func GetMigrationImageBuildContexts(allServiceRequirements []*ServiceRequirement
 	return imageBuildContexts, nil
 }
 
+func checkConflictingMigrations(allDatabases []map[string]*resourcespb.SqlDatabaseResource, resource map[string]*resourcespb.SqlDatabaseResource) error {
+	for _, dbs := range allDatabases {
+		for databaseName, dbConfig := range resource {
+			if existing, exists := dbs[databaseName]; exists {
+				if existing.Migrations.GetMigrationsPath() != dbConfig.Migrations.GetMigrationsPath() {
```

Comment:
Is there a chance dbConfig.Migrations is nil ? might panic here - but not sure if it should error out or continue to process.

            if dbConfig.Migrations == nil {
                continue // or decide if this is an error
            }